File: 3090-maximum-length-substring-with-two-occurrences/3090-maximum-length-substring-with-two-occurrences.cpp
#include <iostream>
#include <string>
#include <algorithm>

using namespace std;

class Solution {
public:
    int maximumLengthSubstring(string s) {
        int n = s.length();
        int max_len = 0;

        for (int i = 0; i < n; i++) {
            // Har naye start index 'i' ke liye 26 size ka clean array banao
            int freq[26] = {0};

            for (int j = i; j < n; j++) {
                int idx = s[j] - 'a'; // Character index (0 to 25)
                freq[idx]++;

                // Agar kisi character ka count 2 se bada hua toh break kar do
                if (freq[idx] > 2) {
                    break;
                }

                // Maximum length update karo
                max_len = max(max_len, j - i + 1);
            }
        }

        return max_len;
    }
};

File: 3090-maximum-length-substring-with-two-occurrences/3090-maximum-length-substring-with-two-occurrences.java
class Solution {
    public int maximumLengthSubstring(String s) {
        int n = s.length();
        int maxLen = 0;
        int left = 0;
        
        // Character counts track karne ke liye 26 size ka simple array
        int[] freq = new int[26];

        for (int right = 0; right < n; right++) {
            int charIdx = s.charAt(right) - 'a';
            freq[charIdx]++;

            // Agar kisi char ka count 2 se bada ho gaya, 
            // toh left pointer ko aage badao jab tak count valid na ho jaye
            while (freq[charIdx] > 2) {
                freq[s.charAt(left) - 'a']--;
                left++;
            }

            // Current valid window ki maximum length update karo
            maxLen = Math.max(maxLen, right - left + 1);
        }

        return maxLen;
    }
}